packages/aws-cdk-lib/aws-rds/lib/aurora-cluster-instance.ts

```diff
@@ -0,0 +1,529 @@
+   */
+  readonly performanceInsightEncryptionKey?: kms.IKey;
+
+  /**
+   * Indicates whether the DB instance is an internet-facing instance.
+   *
+   * @default - true if the instance is placed in a public subnet
+   */
+  readonly publiclyAccessible?: boolean;
+
+  /**
+   * The parameters in the DBParameterGroup to create automatically
+   *
+   * You can only specify parameterGroup or parameters but not both.
+   * You need to use a versioned engine to auto-generate a DBParameterGroup.
+   *
+   * @default - None
+   */
+  readonly parameters?: { [key: string]: string };
+
+  /**
+   * Whether to allow upgrade of major version for the DB instance.
+   *
+   * @default - false
+   */
+  readonly allowMajorVersionUpgrade?: boolean;
+
+  /**
+   * The DB parameter group to associate with the instance.
+   * This is only needed if you need to configure different parameter
+   * groups for each individual instance, otherwise you should not
+   * provide this and just use the cluster parameter group
+   *
+   * @default the cluster parameter group is used
+   */
+  readonly parameterGroup?: IParameterGroup;
+}
+
+/**
+ * Create an RDS Aurora Cluster Instance. You can create either provisioned or
+ * serverless v2 instances.
+ *
+ * @example
+ *
+ * declare const vpc: ec2.Vpc;
+ * const cluster = new rds.DatabaseCluster(this, 'Database', {
+ *   engine: rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_2_08_1 }),
+ *   writer: rds.ClusterInstance.provisioned('writer', {
+ *     instanceType: ec2.InstanceType.of(ec2.InstanceClass.R6G, ec2.InstanceSize.XLARGE4),
+ *   }),
+ *   serverlessV2MinCapacity: 6.5,
+ *   serverlessV2MaxCapacity: 64,
+ *   readers: [
+ *     // will be put in promotion tier 1 and will scale with the writer
+ *     rds.ClusterInstance.serverlessV2('reader1', { scaleWithWriter: true }),
+ *     // will be put in promotion tier 2 and will not scale with the writer
+ *     rds.ClusterInstance.serverlessV2('reader2'),
+ *   ]
+ *   vpc,
+ * });
+ */
+export class ClusterInstance implements IClusterInstance {
+  /**
+   * Add a provisioned instance to the cluster
+   *
+   * @example
+   * ClusterInstance.provisioned('ClusterInstance', {
+   *   instanceType: ec2.InstanceType.of(ec2.InstanceClass.R6G, ec2.InstanceSize.XLARGE4),
+   * });
+   */
+  public static provisioned(id: string, props: ProvisionedClusterInstanceProps = {}): IClusterInstance {
+    return new ClusterInstance(id, {
+      ...props,
+      instanceType: ClusterInstanceType.provisioned(props.instanceType),
+    });
+  }
+
+  /**
+   * Add a serverless v2 instance to the cluster
+   *
+   * @example
+   * ClusterInstance.serverlessV2('ClusterInstance', {
+   *   scaleWithWriter: true,
+   * });
+   */
+  public static serverlessV2(id: string, props: ServerlessV2ClusterInstanceProps = {}): IClusterInstance {
+    return new ClusterInstance(id, {
+      ...props,
+      promotionTier: props.scaleWithWriter ? 1 : 2,
```

Comment:
I thought about this and I decided to do it for provisioned instances, but for serverless I decided not to. The reason is that all serverless instances in the cluster have the same min/max capacity. The only difference is whether it is in tier 0-1 or 2-15. 0-1 means it will scale with the writer and 2-15 means it will not.

I think it would be easy enough to add in later if a use case comes up, but I can't think of one right now.